resources/aws_ec2.go

```diff
@@ -486,6 +491,72 @@ func (obj *AwsEc2Res) longpollWatch() error {
 	}
 }
 
+// snsWatch uses amazon cloudwatch events and simple notification service to
+// detect ec2 instance state changes.
+func (obj *AwsEc2Res) snsWatch() error {
+	send := false
+	var exit *error
+	defer obj.wg.Wait()
+	defer close(obj.closeChan)
+	// set up the sns endpoint
+	snsServer := obj.snsServer()
+	// shutdown the sns endpoint when we're done
+	defer func() {
+		ctx, cancel := context.WithTimeout(context.TODO(), SnsServerTimeout*time.Second)
+		defer cancel()
+		if err := snsServer.Shutdown(ctx); err != nil {
+			if err != context.Canceled {
+				log.Printf("%s: error stopping sns endpoint: %s", obj, err)
+				return
+			}
+			log.Printf("%s: sns server shutdown timed out", obj)
```

Comment:
should be server shutdown `cancelled` instead of timed out, since context might get cancelled by function.